cluster/src/main/java/org/apache/iotdb/cluster/log/applier/AsyncDataLogApplier.java

```diff
@@ -131,7 +132,10 @@ private PartialPath getPlanKey(PhysicalPlan plan) throws StorageGroupNotSetExcep
 
   private PartialPath getPlanSG(PhysicalPlan plan) throws StorageGroupNotSetException {
     PartialPath sgPath = null;
-    if (plan instanceof InsertPlan) {
+    if (plan instanceof InsertMultiTabletPlan) {
+      PartialPath deviceId = ((InsertMultiTabletPlan) plan).getFirstDeviceId();
+      sgPath = IoTDB.metaManager.getStorageGroupPath(deviceId);
+    } else if (plan instanceof InsertPlan) {
```

Comment:
Applying InertMultiTabletPlan parallelly with other insertions is dangerous, for example:
There are two plans, one InsertMultiTabletPlan and one InsertTabletPlan, and the InsertMultiTabletPlan contains 2 insertions: insert timestamp=0,value=0 to root.sg1.s0 and insert timestamp=0,value=0 to root.sg2.s0; while the InsertTabletPlan has only one insertion: insert timestamp=0,value=100 to root.sg2.s0. 
Because the first deviceId is used as the plan key, the two plans may be executed in parallel, so the actual order is not guaranteed on different replicas, that is, on NodeA, InsertMultiTabletPlan may be executed lastly so root.sg2.s0 is 0 at timestamp=0, but on NodeB, InsertTabletPlan is executed later so root.sg2.s0 is 100 at timestamp=0, thus data becomes inconsistent.